thunder/__init__.py

```diff
@@ -571,14 +572,6 @@ def get_computation_and_inputs(*args, **kwargs):
             computation_trc = dce(computation_trc)
             computation_traces.append(computation_trc)
 
-            if is_autocast_enabled:
```

Comment:
Before this PR it was possible to apply autocast with a context manager to any symbol with a registered autocast rule.
Now it's only for functions decorated with "torchsymbol" *and* with a registered autocast rule.